libvast/vast/default_table_slice.hpp

```diff
@@ -63,4 +42,24 @@ class default_table_slice::builder : public table_slice::builder {
   default_table_slice_ptr slice_;
 };
 
+} // namespace detail
+
+/// The default implementation of `table_slice`.
+class default_table_slice : public table_slice {
```

Comment:
Since you are overriding member functions as `final`: maybe the class itself should be `final`?